independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializer.java

```diff
@@ -300,15 +305,43 @@ public void add(ModelProblemCollectorRequest req) {
             remotes.add(new RemoteRepository.Builder(DEFAULT_REMOTE_REPO_ID, "default", DEFAULT_REMOTE_REPO_URL)
                     .setReleasePolicy(new RepositoryPolicy(true, RepositoryPolicy.UPDATE_POLICY_DAILY, RepositoryPolicy.CHECKSUM_POLICY_WARN))
                     .setSnapshotPolicy(new RepositoryPolicy(false, RepositoryPolicy.UPDATE_POLICY_DAILY, RepositoryPolicy.CHECKSUM_POLICY_WARN))
+                    .setProxy(proxy)
                     .build());
         }
 
         return remotes;
     }
 
+    /**
+     * Convert a {@link org.apache.maven.settings.Proxy} to a {@link Proxy}.
+     * @param proxy The proxy to use, may be {@code null}.
+     * @param proxy Maven proxy settings, may be {@code null}.
+     * @return Aether repository proxy or {@code null} if given {@link org.apache.maven.settings.Proxy} is {@code null}.
+     */
+    private static Proxy toAetherProxy(org.apache.maven.settings.Proxy proxy) {
+        if (proxy == null) {
+            return null;
+        }
+        Authentication auth = null;
+        if (proxy.getUsername() != null) {
+            auth = new Authentication() {
+                @Override
+                public void fill(AuthenticationContext context, String key, Map<String, String> data) {
+                    context.put(proxy.getUsername(), proxy.getPassword());
+                }
+
+                @Override
+                public void digest(AuthenticationDigest digest) {
+                    digest.update(proxy.getUsername(), proxy.getPassword());
+                }
+            };
+        }
```

Comment:
I think you are missing another one for proxy.getPassword().
Let's do it the way it's done in newSession(...) and actually call the new method from newSession(...) and getRemoteRepos(...).
Otherwise, it looks great. Thanks!

Reply:
You are still not calling the new method from newSession(...). That's the last bit :)